IGHTMLQuery/Classes/IGXMLNodeQuery.h

```diff
@@ -19,7 +19,7 @@
  @param xpath used to query the document
  @return elements matched by supplied XPath query.
  */
-- (IGXMLNodeSet*) queryWithXPath:(NSString*)xpath;
+- (nullable IGXMLNodeSet*) queryWithXPath:(nullable NSString*)xpath;
```

Comment:
Conceptually I agree with you.

The reason I did that is because IN THEORY it's possible that the alloc/init of a `IGXMLNodeSet` could return nil. Given the way things are presently implemented, the ONLY way that could happen would be if `-[NSObject init]` (the invocation of `super` in `IGXMLNodeSet`'s init method) would fail. That's extremely unlikely, and chances are if THAT happens you're in bigger trouble anyways.

What do you think?